blog/views.py

```diff
@@ -0,0 +1,21 @@
+from django.urls import reverse_lazy
+from django.views import generic
+
+from blog.models import Post, Commentary
+
+
+class Index(generic.ListView):
+    model = Post
+    context_object_name = "post_list"
+    template_name = "blog/index.html"
+    paginate_by = 5
+
+
+class PostDetailView(generic.DetailView):
+    model = Post
+
+
+class CommentCreateView(generic.CreateView):
+    model = Commentary
+    fields = "__all__"
+    success_url = reverse_lazy("blog:index")
```

Comment:
you should   associate the comment with the user who is currently logged in (i.e., the user who sent the request), you can override the form_valid method in your `CommentCreateView`  In this method, you can set the user of the comment before saving it.